src/pages/dataElementGroups/Edit.tsx

```diff
@@ -0,0 +1,191 @@
+import { useDataEngine, useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { NoticeBox } from '@dhis2/ui'
+import { FORM_ERROR, FormApi } from 'final-form'
+import React, { useEffect, useRef } from 'react'
+import { withTypes } from 'react-final-form'
+import { useNavigate, useParams } from 'react-router-dom'
+import {
+    Loader,
+    StandardFormActions,
+    StandardFormSection,
+} from '../../components'
+import { SCHEMA_SECTIONS, getSectionPath } from '../../lib'
+import { JsonPatchOperation } from '../../types'
+import { DataElementGroup } from '../../types/generated'
+import { createJsonPatchOperations } from './edit/'
+import classes from './Edit.module.css'
+import { DataElementGroupFormFields, validate } from './form'
+import type { FormValues } from './form'
+
+type FinalFormFormApi = FormApi<FormValues>
+
+const { Form } = withTypes<FormValues>()
+
+type DataElementGroupQueryResponse = {
+    dataElementGroup: DataElementGroup
+}
+
+const listPath = `/${getSectionPath(SCHEMA_SECTIONS.dataElementGroup)}`
+
+function useDataElementGroupQuery(id: string) {
+    const DATA_ELEMENT_QUERY = {
+        dataElementGroup: {
+            resource: `dataElementGroups/${id}`,
+            params: {
+                fields: ['*', 'attributeValues[*]'],
+            },
+        },
+    }
+
+    return useDataQuery<DataElementGroupQueryResponse>(DATA_ELEMENT_QUERY, {
+        variables: { id },
+    })
+}
+
+function usePatchDirtyFields() {
```

Comment:
This is also basically identical to the one for `dataElements`. Abstracting these to a generic hook would be better imo. It could potentially just take the id and operations as a parameters (instead of calling `createJsonPatchOperations` here) as those can be different for each object - however I do believe most of this can be made generically.

Almost every models in the maintenance app are metadata, and they are very similar in terms of functionality. Think of the models as subclasses of a "metadata"-class, there's very little difference needed in terms of logic/orchestration for creating objects. Mostly we want more "customisability" (compared to old maintenance app) in terms of forms, not for every core logic like this. We should think about escape-hatches - which should be pretty easy with how we've orchestrated our routes. This is why I'm so adamant that we need to approach this more generically. I believe almost everything except for the forms themselves can be generic for every model. 

Currently this does not support or account for attributes. And attributes can be on almost every metadata object. It's not feasible to duplicate this logic for every model.

Again, I'm not sure i's feasible to do that in this PR before release. But it's something we should keep in mind, and take a look at before we implement a ton of models like this.

To give you an idea how generic the old app is, `visualisations` is not part of that app, but its a metadata object and you can actually show a list of by creating an URL like this: https://debug.dhis2.org/dev/dhis-web-maintenance/index.html#/list/visualizationSection/visualization. The list works, editing the identifiable part of the object works, sharing works. I'm not saying this is particularly great, and we should do it like that. But it shows how similar these objects are. The old app has great ideas and implementation in the core logic of handling metadata, but the annoying part is how intertwined that is with the UI. We need to have more customisable UI for each model, but we should still try to handle these things generically where we can. 

This is why I've tried to go the "default"-rute. Create a default way of handling most of the logic, and make it possible to override or change the behaviour, if some model deviates from the default.